game.cpp

```diff
@@ -92,26 +94,35 @@ void game::do_action(player& player, action_type action)
       case action_type::turn_left:
       {
         player.turn_left();
+        player.set_action(action_type::turn_left);
         break;
       }
       case action_type::turn_right:
       {
         player.turn_right();
+        player.set_action(action_type::turn_right);
         break;
       }
-      case action_type::accelerate:
-      {
-        player.accelerate();
-        break;
-      }
-      case action_type::brake:
+      case action_type::accelerate_forward:
       {
-        player.brake();
+        if (player.get_speed() >= 0) {
+          player.accelerate_forward();
+          player.set_action(action_type::accelerate_forward);
```

Comment:
I am unsure if you need to do both modify the player, as well as setting the action (so that the player will do the action at the next tick). Maybe this is a duplicate of things happening that needs to be cleaned in the future, but for now, I'll accept as it passes all tests :-)

Reply:
`set_action()` here will just set a player object's private member `m_action` to `action_type::accelerate_forward`, this will not let the player to do this action at the next tick. It will just give a player object a sign/flag. I do this because I want to detect what action the player is executing at the current tick, to be more specific, I want to detect what key is being pressed at the current tick.

Let's say if `W` is being pressed at the current tick, then the player 0 should accelerate, and `game::apply_inertia()` should be able to detect this fact and thus it won't make the player to decelerate due to the inertia which counteracts with the key `W` being pressed.

Reply:
Maybe we need to refactor this method to make it self-explained? Something like `set_action_flag()`

Reply:
Excellent, I justed wanted to know if you gave it any thought or just randomly did stuff. You have had an idea what you were doing. That is enough for me! Refactoring the method to be more self-explanatory is great, especially after merging this PR to develop :innocent: